manuscripts2/metrics/git.py

```diff
@@ -20,50 +20,82 @@
 #
 # Author:
 #   Pranjal Aswani <aswani.pranjal@gmail.com>
-#
-
-import sys
-sys.path.insert(0, '..')
 
 from manuscripts2.elasticsearch import Query
 
 
 class GitMetrics():
+    """Root of all metric classes based on queries to a git enriched index.
+
+    This class is not intended to be instantiated, but to be
+    extened by child classes that will populate self.query with real
+    queries.
+
+    :param index: index object
+    """
+
+    def __init__(self, index, start, end):
+
+        self.query = Query(index)
+        self.start = start
+        self.end = end
+        self.query.since(self.start).until(self.end)
+
+    def timeseries(self):
+        """Obtain a time series from the current query"""
+
+        return self.query.get_timeseries(dataframe=True)
+
+    def aggregations(self):
+        """Obtain a single valued aggregation from the current query"""
+
+        return self.query.get_aggs()
+
+
+class Commits(GitMetrics):
+    """Class for computing the "commits" metric.
+
+    :param index: index object
+    """
+
+    def __init__(self, index, start, end):
+        super().__init__(index, start, end)
+        self.id = "commits"
+        self.name = "Commits"
+        self.desc = "Changes to the source code"
+        self.query = self.query.get_cardinality("hash").by_period()
```

Comment:
Why not using `query.timeseries` here?

Reply:
@jgbarah we have to add the aggregations into the query object to get the timeseries. So, I am adding the aggregations here and I calculate the timeseries later when I call the `timeseries()` method in report.py.

Reply:
@jgbarah You mean to say that I should calculate the timeseries data here?

Right now I am just adding all the necessary aggregations into the query object and then I get the timeseries in report.py by calling the `timeseries()` method on an instance of the `Commits` class.

Reply:
I think I overlooked the code then, sorry. Now I realize what you do. OK, let's go this way.